SwiftUI/Model/ZimMigration.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ZimMigration.swift
+//  Kiwix
+
+import Foundation
+import CoreData
+
+enum ZimMigration {
+
+    /// Holds the new zimfile host:
+    /// Set during migration,
+    /// and read back when updating URLS mapped from WebView interaction state,
+    /// witch is saved as Data for each opened Tab
+    @MainActor private static var newHost: String?
+    private static let sortDescriptors = [NSSortDescriptor(keyPath: \ZimFile.created, ascending: true)]
+    private static let requestLatestZimFile = ZimFile.fetchRequest(
+        predicate: ZimFile.Predicate.isDownloaded,
+        sortDescriptors: Self.sortDescriptors
+    )
+
+    static func forCustomApps() async {
+        guard FeatureFlags.hasLibrary == false else { return }
+        await Database.shared.container.performBackgroundTask { context in
+            guard var zimFiles = try? requestLatestZimFile.execute(),
+                  zimFiles.count > 1,
+                  let latest = zimFiles.popLast() else {
+                return
+            }
+
+            context.mergePolicy = NSMergePolicy.mergeByPropertyObjectTrump
+            for zimFile in zimFiles {
+                migrateFrom(zimFile: zimFile, toZimFile: latest, using: context)
+            }
+        }
+        debugPrint("migrationForCustomApps Complete")
```

Comment:
Is this supposed to stay?

Reply:
Thanks, good catch, I've removed it now.